src/__tests__/metatests.js

```diff
@@ -73,3 +74,109 @@ describe('jsdom-global', () => {
     }
   });
 });
+
+describe('lolex', () => {
+  let lolex: Lolex;
+
+  beforeAll(() => {
+    lolex = new Lolex();
+  });
+
+  afterAll(() => {
+    lolex.dispose();
+  });
+
+  afterEach(() => {
+    // clear any unset timers
+    lolex.clearAllTimers();
+  });
+
+  test('Date.now() is mocked by Lolex', () => {
+    const start = Date.now();
+    lolex.advanceTimersByTime(5.5e9); // ~63.66 days
+    const end = Date.now();
+
+    const apparentDuration = end - start;
+
+    expect(apparentDuration).toBeGreaterThan(5e9);
+    expect(apparentDuration).toBeLessThan(6e9);
+  });
+
+  test('setInterval is triggered by Lolex', () => {
+    let count = 0;
+    setInterval(() => {
+      ++count;
+    }, 1e9);
+
+    lolex.advanceTimersByTime(5.5e9);
+
+    expect(count).toBe(5);
+  });
+
+  test('setTimeout is triggered by Lolex', () => {
+    let flag = false;
+    setTimeout(() => {
+      flag = true;
+    }, 3e9);
+    setTimeout(() => {
+      flag = true;
+    }, 6e9);
+
+    lolex.advanceTimersByTime(5.5e9);
+
+    expect(flag).toBe(true);
+  });
+
+  test('timer count is properly maintained', () => {
+    expect(lolex.getTimerCount()).toBe(0);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+    setInterval(() => {}, 10);
+
+    expect(lolex.getTimerCount()).toBe(5);
+
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+    setTimeout(() => {}, 10);
+
+    expect(lolex.getTimerCount()).toBe(10);
+
+    lolex.advanceTimersByTime(20);
+    expect(lolex.getTimerCount()).toBe(5);
+
+    lolex.clearAllTimers();
+    expect(lolex.getTimerCount()).toBe(0);
+  });
+
+  test('runOnlyPendingTimers runs timers as expected', () => {
+    const start = Date.now();
+
+    // set up eternal interval
+    let count = 0;
+    const interval = 10;
+    setInterval(() => ++count, interval);
+
+    const maxTime = 1000;
+
+    /* set up several timeouts, of length at most `maxTime` */ {
```

Comment:
nit: please put this open-brace on its own line, not after a comment :-)

Or just delete the braces -- I don't think they really add to the clarity here.

Reply:
As intended! This style of block-comment-plus-brace-pair is useful with code-folding.

That said, this block has been totally rewritten since it got that wrapper (it used to be a manually-specified list, rather than a for-loop), so folding it away is no longer helpful. Braces removed.

Reply:
> useful with code-folding.

Hmm interesting!

I do like having the code be clear when read as-is, though. :wink: And a lot of what makes that work is expectations, and when I see a line-long comment, having an important bit of live code structure slip in at the end of the line is highly unexpected.

Anyway, happy it's moot in this case. If another case comes up in the future, we can discuss more in depth.